colossalai/booster/plugin/hybrid_parallel_plugin.py

```diff
@@ -999,34 +999,37 @@ def __init__(
         ), f"World size {dist.get_world_size()} is not divisible by tp_size {tp_size} * pp_size {pp_size}"
 
         if enable_sequence_parallelism:
-            self.sequence_parallelism_mode = sequence_parallelism_mode if sequence_parallelism_mode is not None else "1"
+            self.sequence_parallelism_mode = (
+                sequence_parallelism_mode if sequence_parallelism_mode is not None else "all_to_all"
+            )
             assert (
                 self.sequence_parallelism_mode in SUPPORT_SP_MODE
             ), f"Sequence parallelism mode {self.sequence_parallelism_mode} is not in the supported list {SUPPORT_SP_MODE}"
             if self.sequence_parallelism_mode in ["split_gather", "ring"]:
-                assert (
-                    tp_size > 1
-                ), f"Sequence parallelism mode {self.sequence_parallelism_mode} must be enabled when using tensor parallelism"
-                if sp_size != 1:
-                    warnings.warn(
-                        f"The sp_size will be the same as tp_size in sequence parallelism mode {self.sequence_parallelism_mode}, will ignore the given sequence parallelism size."
-                    )
-                self.sp_size = 1
-                self.dp_size = dist.get_world_size() // (tp_size * pp_size)
+                # assert (
+                #     tp_size > 1
+                # ), f"Sequence parallelism mode {self.sequence_parallelism_mode} must be enabled when using tensor parallelism"
+                # if sp_size != 1:
+                #     warnings.warn(
+                #         f"The sp_size will be the same as tp_size in sequence parallelism mode {self.sequence_parallelism_mode}, will ignore the given sequence parallelism size."
+                #     )
+                # self.sp_size = 1
+                self.sp_size = sp_size if sp_size is not None else 1
+                self.dp_size = dist.get_world_size() // (tp_size * pp_size * self.sp_size)
```

Comment:
When sp mode is "split_gather" or "ring", it's actually 3d parallelism, as tp and sp share the same process group. The sp size should be always 1 here. If you want to enable "split_gather" and "ulysses" together, you should refactor linear layer.